### PR TITLE
feat(ga): add a new datasource to get the list of the endpoints

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new datasource to get the list of the endpoints.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new datasource to get the list of the endpoints.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (dev_point)$ make testacc TEST="./huaweicloud/services/acceptance/ga" TESTARGS="-run TestAccDatasourceEndpoints_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccDatasourceEndpoints_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEndpoints_basic
=== PAUSE TestAccDatasourceEndpoints_basic
=== CONT  TestAccDatasourceEndpoints_basic
--- PASS: TestAccDatasourceEndpoints_basic (190.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        191.015s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/ga/data_source_huaweicloud_ga_endpoints_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  value = alltrue(local.resource_id_filter_result) && length(local.resource_id_filter_result) > 0
+}
+
+locals {
+  resource_type = data.huaweicloud_ga_endpoints.test.endpoints[0].resource_type
+}
+
+data "huaweicloud_ga_endpoints" "filter_by_resource_type" {
+  endpoint_group_id = huaweicloud_ga_endpoint_group.test.id
+  resource_type       = local.resource_type
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/ga/data_source_huaweicloud_ga_endpoints_test.go:153]: wrong format

Comment (ruwenqiang123) [reply huaweicloud/services/acceptance/ga/data_source_huaweicloud_ga_endpoints_test.go]: done

**Comment by ruwenqiang123** [comment]
```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (dev_point)$ make testacc TEST="./huaweicloud/services/acceptance/ga" TESTARGS="-run TestAccDatasourceEndpoints_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccDatasourceEndpoints_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEndpoints_basic
=== PAUSE TestAccDatasourceEndpoints_basic
=== CONT  TestAccDatasourceEndpoints_basic
--- PASS: TestAccDatasourceEndpoints_basic (188.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        188.525s
```

**Pull request merged**